mmv1/products/compute/FirewallPolicyWithRules.yaml

```diff
@@ -0,0 +1,450 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: FirewallPolicyWithRules
+min_version: beta
+base_url: 'locations/global/firewallPolicies?parentId={{parent}}'
+self_link: 'locations/global/firewallPolicies/{{policy_id}}'
+create_url: 'locations/global/firewallPolicies?parentId={{parent}}'
+update_verb: :PATCH
+description: "The Compute FirewallPolicy with rules resource"
+id_format: 'locations/global/firewallPolicies/{{policy_id}}'
+import_format: ['locations/global/firewallPolicies/{{policy_id}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'compute_firewall_policy_with_rules_full'
+    primary_resource_id: 'firewall-policy-with-rules'
+    vars:
+      policy_name: 'tf-fw-org-policy-with-rules'
+      address_group_name: 'tf-address-group'
+      security_profile_group_name: 'tf-security-profile-group'
+      security_profile_name: 'tf-security-profile'
+    test_env_vars:
+      org_id: :ORG_ID
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  constants: templates/terraform/constants/resource_compute_firewall_policy_with_rules.go.erb
+  encoder: templates/terraform/encoders/resource_compute_firewall_policy_with_rules.go.erb
+  update_encoder: templates/terraform/update_encoder/resource_compute_firewall_policy_with_rules.go.erb
+  decoder: templates/terraform/decoders/resource_compute_firewall_policy_with_rules.go.erb
+  post_create: templates/terraform/post_create/resource_compute_firewall_policy_with_rules.go.erb
+  post_update: templates/terraform/post_update/resource_compute_firewall_policy_with_rules.go.erb
+  post_delete: templates/terraform/post_delete/resource_compute_firewall_policy_with_rules.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: parent
+    description: |
+      The parent of this FirewallPolicy in the Cloud Resource Hierarchy.
+      Format: organizations/{organization_id} or folders/{folder_id}
+    required: true
+    immutable: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: creationTimestamp
+    description: Creation timestamp in RFC3339 text format.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: shortName
+    description: A textual name of the security policy.
+    immutable: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: policyId
+    description: The unique identifier for the resource. This identifier is defined by the server.
+    output: true
+    api_name: id
+  - !ruby/object:Api::Type::String
+    name: description
+    description: An optional description of this resource.
+  - !ruby/object:Api::Type::Array
+    name: 'rule'
+    api_name: 'rules'
+    description: A list of firewall policy rules. It must include pre-defined rules.
```

Comment:
Does it need to include pre-defined rules? Don't they get split later?

Reply:
right, fixed.